crypto/crypto/src/lib.rs

```diff
@@ -49,3 +54,9 @@ compile_error!(
     "at most one dalek arithmetic backend cargo feature should be enabled! \
      please enable one of: fiat, vanilla"
 );
+
+// MIRAI's tag analysis makes use of the incomplete const_generics feature, so the module
+// containing the definitions of MIRAI tag types should not get compiled in a release build.
+// The code below fails a build of the crate if mirai is on but debug_assertions is not.
+#[cfg(all(mirai, not(debug_assertions)))]
+compile_error!("MIRAI can only be used to compile the crate in a debug build!");
```

Comment:
is it really the right place for this assertion?

Reply:
It would be better if the assertion is implemented universally for the project. Any suggestions?